BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Authentication/ICredentialEnvelope.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Authentication
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public interface ICredentialEnvelope
```

Comment:
Do we need this to be public? Consider hiding it since we don't expect external components to take dependency.
Probably we don't need an interface at all?

Reply:
I think this NEEDS to be public so other parts of our SDK can access the GetToken method.
Reminder that this will be accessible from the `TelemetryConfiguration`. 
This will be in the Microsoft.ApplicationInsights library, but will need to be accessed by ServerTelemetryChannel and QuickPulseTelemetryModule which are both in separate modules. Also if SnapshotDebugger and Profiler intend to leverage our work here.

Reply:
Can you comment on why you would recommend an abstract class here?
I went back and forth on this exact topic, but ultimately I couldn't find a use case for the abstract class.
Maybe you're considering something here that I've overlooked and I would like to learn :)

Reply:
`public interface ICredentialEnvelope` means anyone could take dependency and implement that interface, which means we cannot add new stuff since it breaks compatibility.